internal/services/dataprotection/data_protection_backup_policy_mysql_flexible_server_resource.go

```diff
@@ -0,0 +1,705 @@
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"criteria": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"absolute_criteria": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForAbsoluteMarker(), false),
+								},
+
+								"days_of_week": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForDayOfWeek(), false),
+									},
+								},
+
+								"months_of_year": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForMonth(), false),
+									},
+								},
+
+								"scheduled_backup_times": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.IsRFC3339Time,
+									},
+								},
+
+								"weeks_of_month": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForWeekNumber(), false),
+									},
+								},
+							},
+						},
+					},
+
+					"life_cycle": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"data_store_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										// Confirmed with the service team that currently only `VaultStore` is supported.
+										// However, considering that `ArchiveStore` will be supported in the future, it would be exposed for user specification.
+										string(backuppolicies.DataStoreTypesVaultStore),
+									}, false),
+								},
+
+								"duration": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: azValidate.ISO8601Duration,
+								},
+							},
+						},
+					},
+
+					"priority": {
```

Comment:
does it have a value range? Can it be a minus value?

Reply:
All back policies for all azure services are almost same in https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection. You may still remember you reviewed the https://github.com/hashicorp/terraform-provider-azurerm/pull/26024 for Backup Policy for Postgresql fs before. I referred to all backup polices in https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection to make this PR. So I think we may need to align the style.